pkg/scheduler/scheduler.go

```diff
@@ -248,7 +248,7 @@ func (s *Scheduler) schedule(ctx context.Context) wait.SpeedSignal {
 			if len(e.preemptionTargets) != 0 {
 				// If preemptions are issued, the next attempt should try all the flavors.
 				e.LastAssignment = nil
-				preempted, err := s.preemptor.IssuePreemptions(ctx, &e.Info, e.preemptionTargets, cq)
+				preempted, err := s.preemptor.IssuePreemptions(ctx, &e.Info, e.preemptionTargets, cq, e.assignment.Borrowing)
```

Comment:
This feels a bit hacky.

Maybe a better strategy is to calculate the "reason" in `GetTargets`, which already has all the information about the preemptor.

We could have something like this in the preemption package:

```golang
struct Target {
  *workload.Info
  Reason string
  Scope string // ClusterQueue/Cohort
}
```

Then `GetTargets` would return `[]Target`.

Note that `e.assignment.Borrowing` is not enough to determine that it's fair sharing. This is because we have the feature `borrowWithinCohort` which can preempt while borrowing even if fair sharing is enabled. https://kueue.sigs.k8s.io/docs/concepts/cluster_queue/#preemption

And that's all calculated in `GetTargets`.